### PR TITLE
Add initialize_constituents scheme

### PR DESCRIPTION
**Pull request by peverwhee**
Originator(s): peverwhee

Summary (include the keyword ['closes', 'fixes', 'resolves'] and issue number): Adds new (non-portable)  initialize_constituents scheme (.F90 and .meta). This scheme will be used to instantiate all constituents in the file (used to validate schemes that need the full constituent array but do not modify all of the individual constituents). Added to new "test_schemes" directory.

Describe any changes made to the namelist: N/A

List all files eliminated and why: N/A

List all files added and what they do:
A  test/test_schemes/initialize_constituents.F90
A  test/test_schemes/initialize_constituents.F90
 - add initialize_constituents scheme

List all existing files that have been modified, and describe the changes:
(Helpful git command: git diff --name-status development...<your_branch_name>)

List any test failures: none

Is this a science-changing update? New physics package, algorithm change, tuning changes, etc? No

### CODE REVIEW

File: test/test_schemes/initialize_constituents.F90 [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+             long_name = constituent_names(var_index), &
+             units = 'kg kg-1', &
+             vertical_dim = 'vertical_layer_dimension', &
+             min_value = 0.0_kind_phys, &
+             advected = .true., &
+             water_species = .true., &
+             mixing_ratio_type = 'wet', &
+             errcode = errcode, &
+             errmsg = errmsg)
+       else
```
Comment (nusbaume) [inline test/test_schemes/initialize_constituents.F90:113]: I believe we might need an `else if` here that checks if the constituent name also matches any of the names in `const_std_names` as well (which may need to be renamed to something like `water_species_std_names`), and if found will need the `water_species` and `mixing_ratio_type` flags set as well.  In other words:

```suggestion
       else if (any(const_std_names == trim(constituent_names(var_index)))) then
          call constituents(var_index)%instantiate( &
             std_name = constituent_names(var_index), &
             long_name = constituent_names(var_index), &
             units = 'kg kg-1', &
             vertical_dim = 'vertical_layer_dimension', &
             min_value = 0.0_kind_phys, &
             advected = .true., &
             water_species = .true., &
             mixing_ratio_type = 'wet', &
             errcode = errcode, &
             errmsg = errmsg)
       else
```

**Review by nusbaume — CHANGES_REQUESTED** [review]
Thanks @peverwhee for getting this test helper scheme developed!  I just had a few change requests.

File: test/test_schemes/initialize_constituents.F90 [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    ncdata => initial_file_get_id()
+    ! See how many variables are present on the file
+    ierr = pio_inquire(ncdata, nVariables=num_variables)
+    allocate(constituent_names(num_variables), stat=ierr)
+    if (ierr /= 0) then
+       errcode = 1
+       errmsg = 'Failed to allocate "constituent_names"'
+       return
+    end if
```
[the remark this answers is not in the document]
Comment (peverwhee) [reply test/test_schemes/initialize_constituents.F90]: added!

File: test/test_schemes/initialize_constituents.F90 [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+       errmsg = 'Failed to allocate "constituents"'
+       return
+    end if
+
+    do var_index = 1, size(constituents)
+       if (any(water_species_number_concentrations == trim(constituent_names(var_index)))) then
+          call constituents(var_index)%instantiate( &
+             std_name = constituent_names(var_index), &
+             long_name = constituent_names(var_index), &
+             units = 'kg kg-1', &
```
[the remark this answers is not in the document]
Comment (peverwhee) [reply test/test_schemes/initialize_constituents.F90]: updated!

File: test/test_schemes/initialize_constituents.F90 (hunk 1 above)
Comment (peverwhee) [reply test/test_schemes/initialize_constituents.F90:126]: a very good point - added `else if`

**Review by nusbaume — APPROVED** [review]
Everything looks good to me now, thanks!

File: test/test_schemes/initialize_constituents.F90 [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          end if
+       end if
+    end do
+
+    allocate(constituents(constituent_index), stat=ierr, errmsg=alloc_err_msg)
+    if (ierr /= 0) then
+       errcode = 1
+       errmsg = alloc_err_msg
+       return
+    end if
```
[the remark this answers is not in the document]
Comment (peverwhee) [reply test/test_schemes/initialize_constituents.F90]: good call! i've added context to the message.

**Review by cacraigucar — APPROVED** [review]

**Pull request merged**